File: frontend/nda/registry.ts
// Typed access to the generated document registry (see scripts/build-registry.mjs).
// Each entry carries the document's metadata, its fillable fields, and the
// verbatim Standard Terms embedded from templates/.

import registryData from "./registry.json";

export interface TermsSection {
  n: number;
  heading: string;
  blocks: string[];
}

export interface DocumentType {
  id: string;
  name: string;
  description: string;
  terms: TermsSection[];
  attribution: string;
}

export const DOCUMENT_TYPES: DocumentType[] = registryData as DocumentType[];

export function getDocumentType(id: string): DocumentType | undefined {
  return DOCUMENT_TYPES.find((d) => d.id === id);
}

File: backend/app/registry.py
"""Loads the lite document registry (see scripts/build-registry.mjs).

Holds the supported document types with their metadata and fillable fields —
enough to route the user to a document and drive the interview. The verbatim
Standard Terms live only in the frontend registry, since the LLM never writes
legal text.
"""

import json
from pathlib import Path

_REGISTRY_PATH = Path(__file__).resolve().parent / "registry.json"

DOCUMENT_TYPES: list[dict] = json.loads(_REGISTRY_PATH.read_text())

DOCUMENT_IDS: set[str] = {d["id"] for d in DOCUMENT_TYPES}


def get_document_type(doc_id: str) -> dict | None:
    return next((d for d in DOCUMENT_TYPES if d["id"] == doc_id), None)
